test/jdk/javax/swing/plaf/basic/BasicTreeUI/TreeCellRendererLeakTest.java

```diff
@@ -0,0 +1,208 @@
+import javax.swing.tree.TreeNode;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class TreeCellRendererLeakTest {
+
+    static int smCount = 0;
+
+    private static JFrame frame;
+    private JPanel jPanel1;
+    private JPanel jPanel2;
+    private JScrollPane jScrollPane1;
+    private JTabbedPane jTabbedPane1;
+    private JTree jTree1;
+
+    static CountDownLatch testDone;
+
+    ArrayList<WeakReference<JLabel>> weakRefArrLabel = new ArrayList(50);
+
+    // Custom TreeCellRenderer
+    public class TreeCellRenderer extends DefaultTreeCellRenderer {
+
+        public TreeCellRenderer() {}
+
+        // Create a new JLabel every time
+        public Component getTreeCellRendererComponent(
+                JTree tree,
+                Object value,
+                boolean sel,
+                boolean expanded,
+                boolean leaf,
+                int row,
+                boolean hasFocus) {
+            JLabel label = new JLabel();
+            label.setText("TreeNode: " + value.toString());
+            if (sel) {
+                label.setBackground(getBackgroundSelectionColor());
+            } else {
+                label.setBackground(getBackgroundNonSelectionColor());
+            }
+
+            weakRefArrLabel.add(smCount++, new WeakReference<JLabel>(label));
+            return label;
+        }
+    }
+
+    public TreeCellRendererLeakTest() {
+        initComponents();
+        jTree1.setCellRenderer(new TreeCellRenderer());
+        Thread updateThread = new Thread(this::runChanges);
+        updateThread.setDaemon(true);
+        updateThread.start();
+        Thread infoThread = new Thread(this::runInfo);
+        infoThread.setDaemon(true);
+        infoThread.start();
+    }
+
+    // <editor-fold defaultstate="collapsed" desc=" Generated Code ">
+    private void initComponents() {
+        jTabbedPane1 = new javax.swing.JTabbedPane();
+        jPanel1 = new javax.swing.JPanel();
+        jScrollPane1 = new javax.swing.JScrollPane();
+        jTree1 = new javax.swing.JTree();
+        jPanel2 = new javax.swing.JPanel();
+
+        jPanel1.setLayout(new java.awt.BorderLayout());
+
+        jScrollPane1.setViewportView(jTree1);
+
+        jPanel1.add(jScrollPane1, java.awt.BorderLayout.CENTER);
+
+        jTabbedPane1.addTab("tab1", jPanel1);
+
+        jPanel2.setLayout(new java.awt.BorderLayout());
+
+        jTabbedPane1.addTab("tab2", jPanel2);
+
+        jTabbedPane1.setSelectedIndex(1);
+
+        frame = new JFrame();
+        frame.getContentPane().add(jTabbedPane1, java.awt.BorderLayout.CENTER);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }// </editor-fold>
+
+    public static void main(String args[]) throws Exception {
+        try {
+            testDone = new CountDownLatch(1);
+            SwingUtilities.invokeAndWait(() -> {
+                TreeCellRendererLeakTest tf = new TreeCellRendererLeakTest();
```

Comment:
```suggestion
                new TreeCellRendererLeakTest();
```
The variable is unused and can safely be removed. The side effects of calling the constructor won't disappear.